custom_components/adaptive_lighting/switch.py

```diff
@@ -1166,21 +1166,26 @@ async def _adapt_light(
         elif supports_colors and adapt_color:
             _LOGGER.debug("%s: Setting rgb_color of light %s", self._name, light)
             service_data[ATTR_RGB_COLOR] = self._settings["rgb_color"]
-
-        context = context or self.create_context("adapt_lights")
-
-        # See #80. Doesn't check if transitions differ but it does the job.
-        last_service_data = self.turn_on_off_listener.last_service_data
-        if not force and last_service_data.get(light) == service_data:
+        # Check if service data differs from the last. See #80.
+        listener = self.turn_on_off_listener
+        last_service_data = listener.last_service_data.get(light)
+        ignore_fields = {ATTR_TRANSITION}
+        if (
+            not force
+            and last_service_data
+            and {k for k, _ in last_service_data.items() ^ service_data.items()}
```

Comment:
Isn't this the same as `last_service_data.items() ^ service_data.items()`?

Reply:
I believe so. I committed the following change:

(last_service_data.items() ^ service_data.items()) == ignore_fields

Reply:
Oh sorry. I actually didn't see the `== ignore_fields`. The previous thing you had was actually doing something differently (and I believe it was correct before 😅)

Reply:
Yeah the line was too long so the == ignore_fields went to the new line 😄 that caught me off guard too as its the only situation it happens in the entire project.